File: ontrack-extension-svn/src/main/java/net/nemerosa/ontrack/extension/svn/graphql/SVNConfigurationGQLType.java
package net.nemerosa.ontrack.extension.svn.graphql;

import com.google.common.collect.ImmutableSet;
import graphql.schema.GraphQLObjectType;
import net.nemerosa.ontrack.extension.svn.model.SVNConfiguration;
import net.nemerosa.ontrack.graphql.schema.GQLType;
import net.nemerosa.ontrack.graphql.schema.GQLTypeCache;
import net.nemerosa.ontrack.graphql.support.GraphQLBeanConverter;
import org.springframework.stereotype.Component;

@Component
public class SVNConfigurationGQLType implements GQLType {

    @Override
    public String getTypeName() {
        return SVNConfiguration.class.getSimpleName();
    }

    @Override
    public GraphQLObjectType createType(GQLTypeCache cache) {
        return GraphQLBeanConverter.INSTANCE.asObjectType(SVNConfiguration.class, cache, ImmutableSet.of("password", "descriptor", "credentials"));
    }
}

File: ontrack-extension-svn/src/main/java/net/nemerosa/ontrack/extension/svn/graphql/SVNConfigurationGQLType.kt
package net.nemerosa.ontrack.extension.svn.graphql

import graphql.schema.GraphQLObjectType
import net.nemerosa.ontrack.extension.svn.model.SVNConfiguration
import net.nemerosa.ontrack.graphql.schema.GQLType
import net.nemerosa.ontrack.graphql.schema.GQLTypeCache
import net.nemerosa.ontrack.graphql.support.GraphQLBeanConverter
import org.springframework.stereotype.Component

@Component
class SVNConfigurationGQLType : GQLType {

    override fun getTypeName(): String = SVNConfiguration::class.java.simpleName

    override fun createType(cache: GQLTypeCache): GraphQLObjectType {
        return GraphQLBeanConverter.asObjectType(SVNConfiguration::class.java, cache, setOf("password", "descriptor", "credentials"))
    }
}
